frontend/src/modules/entitydetails/components/EntityDetails.js

```diff
@@ -227,9 +227,21 @@ export class EntityDetailsBase extends React.Component<InternalProps, State> {
             pk = selectedEntity.pk;
         }
 
-        dispatch(
-            machinery.actions.get(source, locale, user.isAuthenticated, pk),
-        );
+        if (!pk) {
+            dispatch(
+                machinery.actions.getConcordanceSearchResults(
+                    source,
+                    locale,
+                    page,
+                ),
+            );
+        }
+
+        if (pk || !page) {
```

Comment:
I'm curious, why did you also add `pk ||`? Doesn't seem necessary.

To add more clarity, perhaps we can split this method into `searchMachinery` (concordance search, including loading more results) and `resetMachinery` (reset machinery with a click or when searching empty string). I'll leave the decision to you.

Reply:
TBH, I don't know why I thought that was needed :(

I think splitting it into 2 methods would add clarity. For now I've removed `pk` from the condition, and I will explore splitting this out a bit later today.